src/test/scala/com/github/timgent/dataflare/repository/DfApiQcResultsRepositoryTest.scala

```diff
@@ -0,0 +1,86 @@
+package com.github.timgent.dataflare.repository
+
+import com.github.timgent.dataflare.checks.CheckDescription.SimpleCheckDescription
+import com.github.timgent.dataflare.checks.QcType.{ArbDualDsCheck, ArbSingleDsCheck}
+import com.github.timgent.dataflare.checks._
+import com.github.timgent.dataflare.checkssuite.CheckSuiteStatus.{Error, Success}
+import com.github.timgent.dataflare.checkssuite.{CheckSuiteStatus, ChecksSuiteResult}
+import com.github.timgent.dataflare.utils.CommonFixtures._
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import sttp.client3._
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class DfApiQcResultsRepositoryTest
+    extends AsyncWordSpec
+    with Matchers
+    with Eventually
+    with EsTestUtils
+    with ScalaCheckDrivenPropertyChecks {
+
+  "DfApiResultsRepository.save" should {
+    def generateRawCheckResult(qcType: QcType, suffix: String, status: CheckStatus) =
+      CheckResult(qcType, status, s"checkResult$suffix", SimpleCheckDescription(s"checkDescription$suffix"))
+
+    val uri = uri"http://127.0.0.1:8080"
```

Comment:
We should not post to a real endpoint in unit tests... unless we really want this to be an integration test

Reply:
You can test this one with some mocking API tool (e.g. RestAssured)